qiskit_aer/backends/aerbackend.py

```diff
@@ -84,20 +84,26 @@ def __init__(self,
 
     def _convert_circuit_binds(self, circuit, binds):
         parameterizations = []
-        for index, inst_tuple in enumerate(circuit.data):
-            if inst_tuple[0].is_parameterized():
-                for bind_pos, param in enumerate(inst_tuple[0].params):
+        for index, instruction in enumerate(circuit.data):
+            if instruction.operation.is_parameterized():
+                for bind_pos, param in enumerate(instruction.operation.params):
                     if param in binds:
-                        parameterizations.append([[index, bind_pos], binds[param]])
+                        parameterizations.append([(index, bind_pos), binds[param]])
                     elif isinstance(param, ParameterExpression):
                         # If parameter expression has no unbound parameters
                         # it's already bound and should be skipped
                         if not param.parameters:
                             continue
-                        local_binds = {k: v for k, v in binds.items() if k in param.parameters}
-                        bind_list = [dict(zip(local_binds, t)) for t in zip(*local_binds.values())]
+                        len_vals = len(next(iter(binds.values())))
```

Comment:
Have we checked that `binds` is non-empty before now?  This will fail with a hard-to-understand error if it's empty, which seems like a mistake a user could fairly easily make in a big script file.  If it's not too much overhead, we might also want to test that they're all the same length too, before we start indexing into them.

Reply:
Yes. This line guarantee the dictionary is not empty. https://github.com/Qiskit/qiskit-aer/blob/01b32ce7100164c81e2ffb7e1752bc733c6aacec/qiskit_aer/backends/aerbackend.py#L388

Reply:
Sure, but I don't think that's what we're calling `next` on here; we're doing that on sequences contained within `parameter_binds`, I think?  So we could still be in the position where the user has (erroneously) passed us something like
```python
parameter_binds = [{}, {}, {}]
```
`bool(parameter_binds)` is true, but `binds` in this function is bound to (e.g.) `parameter_binds[0]`, at which point a `next(iter(binds.values()))` will fail with a fairly opaque error.  I haven't tested that though, so I could still be wrong.

Reply:
Ah...You are right. My bad.
<img width="576" alt="image" src="https://user-images.githubusercontent.com/6814928/220134513-5c9542a1-202a-4c6a-b159-5eae9148af0c.png">

But current implementation. Kernel is dead..
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/6814928/220137099-fe830e50-e0d1-4d2f-8aec-b95482a02bbc.png">

We should have a clear message. Thank you.

Reply:
Ah yeah, the current implementation won't have any checks at all, so it'll be getting passed down to C++ and dying with some out-of-bounds access in there.  Your new way is definitely better, having the check in Python space, let's just make sure it has a more clear error (or handles zero parameter binds gracefully).